SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/DynamicSubobjectsTest.cpp

```diff
@@ -0,0 +1,249 @@
+
+					if (IsValid(PlayerCharacter))
+					{
+						if (PlayerCharacter->GetActorLocation().Equals(CharacterRemoteLocation, 1.0f))
+						{
+							FinishStep();
+						}
+					}
+				},
+				StepTimeLimit);
+		}
+
+		{ // Step 5 - Server increases AReplicatedGASTestActor's TestIntProperty to enable checking if the client is out of interest later.
+			AddStep(TEXT("DynamicSubobjectsTestServerIncreasesIntValue"), FWorkerDefinition::Server(1), nullptr, [this, i]() {
+				TArray<AActor*> FoundActors;
+				UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedGASTestActor::StaticClass(), FoundActors);
+				if (FoundActors.Num() == 1)
+				{
+					TestActor = Cast<AReplicatedGASTestActor>(FoundActors[0]);
+					TestActor->TestIntProperty = i + 1;
+				}
+				if (TestActor->TestIntProperty == i + 1)
+				{
+					FinishStep();
+				}
+			});
+		}
+
+		{ // Step 6 - Client 1 checks it can no longer see the AReplicatedDynamicSubobjectsTestActor
+			AddStep(
+				TEXT("DynamicSubobjectsTestClientCheckIntValueIncreased"), FWorkerDefinition::Client(1), nullptr, nullptr,
+				[this, i](float DeltaTime) {
+					TArray<AActor*> FoundActors;
+					UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedGASTestActor::StaticClass(), FoundActors);
+					if (FoundActors.Num() == 1)
+					{
+						TestActor = Cast<AReplicatedGASTestActor>(FoundActors[0]);
+
+						RequireNotEqual_Int(TestActor->TestIntProperty, i + 1, TEXT("Check TestIntProperty didn't get replicated"));
+						StepTimer += DeltaTime;
+						if (StepTimer >= 0.5f)
+						{
+							FinishStep();
+							StepTimer = 0.0f; // reset for the next time
+						}
+					}
+				},
+				StepTimeLimit);
+		}
+
+		{ // Step7 - Server moves Client 1 close to the cube.
+			AddStep(TEXT("DynamicSubobjectsTestServerMoveClient1CloseToCube"), FWorkerDefinition::Server(1), nullptr, [this]() {
+				if (ClientOneSpawnedPawn->SetActorLocation(CharacterSpawnLocation))
+				{
+					if (ClientOneSpawnedPawn->GetActorLocation().Equals(CharacterSpawnLocation, 1.0f))
+					{
+						FinishStep();
+					}
+				}
+			});
+		}
+
+		{ // Step 8 - Client 1 checks that the movement was replicated correctly.
+			AddStep(
+				TEXT("DynamicSubobjectsTestClientCheckSecondMovement"), FWorkerDefinition::Client(1), nullptr, nullptr,
+				[this](float DeltaTime) {
+					ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
+					APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
+					ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+
+					if (IsValid(PlayerCharacter))
+					{
+						if (PlayerCharacter->GetActorLocation().Equals(CharacterSpawnLocation, 1.0f))
+						{
+							FinishStep();
+						}
+					}
+				},
+				StepTimeLimit);
+		}
+
+		{ // Step 9 - Client 1 checks it can see the AReplicatedDynamicSubobjectsTestActor
```

Comment:
Also checks the `TestIntProperty` no?

Reply:
yep thats what I mean by see that you get the replication so you are within the interest.